climakitae/view.py

```diff
@@ -79,92 +79,77 @@ def _visualize(data, lat_lon=True, width=None, height=None, cmap=None):
-                _plot = plt.gcf()  # Add plot to figure
-                plt.close()  # Close to prevent annoying matplotlib collections object line from showing in notebook
-        # If there's more than one grid cell, generate a pretty map
-        else:
-            # Define colorbar label using variable and units
-            try:
-                clabel = data.name + " (" + data.attrs["units"] + ")"
-            except:  # Try except just in case units attribute is missing from data
-                clabel = data.name
-
-            # Set default cmap if no user input
-            # Different if using hvplot (we need "hex")
-            if cmap in [
-                "categorical_cb",
-                "ae_orange",
-                "ae_diverging",
-                "ae_blue",
-                "ae_diverging_r",
-            ]:
-                cmap = _read_ae_colormap(cmap=cmap, cmap_hex=True)
-
-            # Set default width & height
-            if width is None:
-                width = 550
-            if height is None:
-                height = 450
-
-            if set(["x", "y"]).issubset(set(data.dims)):
-                # Reproject data to lat/lon
-                if lat_lon == True:
-                    try:
-                        data = _reproject_data(
-                            xr_da=data, proj="EPSG:4326", fill_value=np.nan
-                        )
-                    except:  # Reprojection can fail if the data doesn't have a crs element. If that happens, just carry on without projection (i.e. don't raise an error)
-                        pass
-
-                # Create map
+                    data = _reproject_data(
+                        xr_da=data, proj="EPSG:4326", fill_value=np.nan
+                    )
+                except:  # Reprojection can fail if the data doesn't have a crs element. If that happens, just carry on without projection (i.e. don't raise an error)
+                    pass
+        if set(["lat", "lon"]).issubset(set(data.dims)):
+            x = "lon"
+            y = "lat"
+
+        # Create map
+        try:
+            if len(data[x]) > 1 and len(data[y]) > 1:
+                # If data has more than one grid cell, make a pretty map
                 _plot = data.hvplot.image(
-                    x="x",
-                    y="y",
+                    x=x,
+                    y=y,
                     grid=True,
                     clabel=clabel,
                     cmap=cmap,
                     width=width,
                     height=height,
                 )
             else:
-                # Create map
-                _plot = data.hvplot.image(
-                    x="lon",
-                    y="lat",
+                # Make a scatter plot if it's just one grid cell
+                print(
+                    "Warning: your input data has 2 or less gridcells. Due to plotting limitations for small areas, a scatter plot will be generated."
```

Comment:
While I'm not a huge fan of the scatter plot, I do think this is an appropriate solution to do a scatter plot on too large resolutions. Could we potentially modify this warning to suggest trying a finer resolution (when reasonable)?